spec/feature/post_index_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.describe 'posts#index', type: :feature do
+  before(:each) do
+    User.destroy_all # Destroy all existing users to start with a clean slate
+    Post.destroy_all # Destroy all existing posts to start with a clean slate
+
+    @user_one = User.create(
+      name: 'Sazzad',
+      photo: 'http://via.placeholder.com/250x250',
+      bio: 'A programmer from Bangladesh.',
+      post_counter: 0
+    )
+
+    @posts = [
+      Post.create(
+        author: @user_one,
+        title: 'Blog1',
+        text: 'Post content 1'
+      ),
+      Post.create(
+        author: @user_one,
+        title: 'Blog2',
+        text: 'Post content 2'
+      )
+    ]
+
+    visit user_posts_url(user_id: @user_one.id)
+  end
+
+  describe '#Indexpage' do
+    it 'can see the user profile picture.' do
+      expect(page).to have_css("img[src='#{@user_one.photo}']")
+    end
+    it 'I can see the user username.' do
+      expect(page).to have_content(@user_one.name.to_s)
+    end
+    it 'I can see the number of posts the user has written.' do
+      expect(page).to have_content(@user_one.post_counter.to_s)
+    end
+
+    it 'should see title of the post' do
+      @posts.each do |post|
+        expect(page).to have_content(post.title.to_s)
+      end
+    end
+
+    it 'should see body of the post' do
+      @posts.each do |post|
+        expect(page).to have_content(post.text.to_s)
+      end
+    end
+
+    it 'I can see the first comments on a post.' do
+      @posts.each do |post|
+        post.recent_comments.each do |comment|
+          expect(page).to have_content(comment.user.text)
+        end
+      end
+    end
+    it 'I can see how many comments a post has' do
+      @posts.each do |post|
+        expect(page).to have_content("Comments:#{post.comment_counter}")
+      end
+    end
+    it 'I can see how many likes a post has' do
+      @posts.each do |post|
+        expect(page).to have_content("Likes:#{post.like_counter}")
+      end
+    end
+  end
+  describe 'GET show/page' do
+    it 'When I click on a post, I am redirected to that postshow page.' do
+      visit user_posts_url(user_id: @user_one.id)
+      post = @posts.first
+      click_link(post.id)
+      expect(page).to have_current_path(user_post_path(@user_one.id, post.id))
+    end
+  end
+end
```

Comment:
- [ ] Great work making all the changes requested by the previous reviewer. However, I noticed that all the tests didn't pass. Kindly fix the issues and ensure that all the tests pass. 👍 

![image](https://github.com/SAZZAD1Q2/Blog-Web/assets/77106962/a6529194-294e-4298-913a-39bac1f8aad0)